design-proposals/project-deploymnet.md

```diff
@@ -0,0 +1,174 @@
+The deployment to another environment as VMs is out of scope.
+* Security considerations related to the creation of certificates for the fabric or each site by the fabric administrator are also beyond the scope of this document
+
+## Proposal
+
+The ClusterLink deployment will be done in the following steps:
+
+1. Deploy the cluster certificates as k8s secrets to the cluster.
+   The certificate include the public Fabric CA certificates and the site certificate that generated by the fabric administrator.
+2. Create the ClusterLink operator.
+3. Deploy the ClusterLink CRD to the operator ClusterLink deployments that includes:
+
+   * ClusterLink controlplane deployment, including controlplane-pod, controlplane-service, and RBAC roles.  
+   * ClusterLink dataplane deployment, including dataplane pods, whether single or multiple, along with the dataplane-service..
+   * ClusterLink external access point using load-balancer/node-port/gateway-API service.
+  
+Overall, the ClusterLink deployment stages are:
+
+<img src="deployment.png" width="800" height="400" alt="ClusterLink Deployment Stages"/>
+
+The operator will be created in a dedicated namespace and, by default, will be ```cl-operator-ns```. It will have privileged permissions, enabling it to create ClusterLink components within the dedicated namespace.
+The ClusterLink components will be created in ```clusterlink-system-ns```, and the control-plane will have privileged permissions to execute operations on the ClusterLink CRDs, such as peer, export, import, policy, etc.
+Additionally, the operator will create and update components in response to the ```clusterlink.yaml``` create or update actions. Once created or updated, the operator will not actively monitor the state of each component, assuming that any changes will be made only by privileged users.
+Furthermore, the operator will delete all components and the namespace of ClusterLink when the ```clusterlink.yaml``` file is deleted.
+
+The deployment file for the operator and example of ```clusterlink.yaml``` will be included in every project release.
+
+The ClusterLink CLI will be utilized to automate the deployment process.\
+For instance, the following commands can assist the fabric administrator in creating certificates in the directory ```~/.config/clustrlink/<fabric_name>```:
+
+    clusterlink create fabric --name <fabric_name>
+    clusterlink create site --name <site_name> --fabric <fabric_name> 
+
+Automation for deploying the certificates as a secret to the cluster, deploying the ClusterLink operator, and the ```clusterlink.yaml``` can be achieved with
+
+    clusterlink install --site <site_name>
+
+### ClusterLink CRD
+
+The ClusterLink CRD includes the following fields:
+
+* Api version: clusterlink.net/v1alpha1
+* Kind: clusterlink
+
+* **Spec Section:**
+  
+    | Field name | Description | Default value |
+    | ---- | ----- | ------ |
+    |dataplane.type| Types of dataplane, supported values "go" or "envoy"|"envoy"|
```

Comment:
I don't think the use of `.` in field names is common. Unless you meant that there's a sub-object?